### PR TITLE
CLOUDP-57303: Improve delete commands structure

### PR DESCRIPTION
**Pull request by gssbzn**
This tries to extract some of the repeating blocks so we can reuse them later

### CODE REVIEW

File: internal/cli/cloud_manager_clusters_shutdown.go [hunk 1]
```diff
@@ -40,17 +40,15 @@ func (opts *cmClustersShutdownOpts) init() error {
...
-
-	opts.store = s
-	return nil
+	var err error
+	opts.store, err = store.New()
+	return err
 }
 
 func (opts *cmClustersShutdownOpts) Run() error {
+	if !opts.confirm {
```
Comment (gssbzn) [inline internal/cli/cloud_manager_clusters_shutdown.go]: fixing a bug that would ignore the answer to confirm

File: internal/cli/cloud_manager_clusters_startup.go [hunk 2]
```diff
@@ -40,17 +40,15 @@ func (opts *cmClustersStartupOpts) init() error {
...
-
-	opts.store = s
-	return nil
+	var err error
+	opts.store, err = store.New()
+	return err
 }
 
 func (opts *cmClustersStartupOpts) Run() error {
+	if !opts.confirm {
```
Comment (gssbzn) [inline internal/cli/cloud_manager_clusters_startup.go]: fixing a bug that would ignore the answer to confirm

File: internal/cli/cli.go [hunk 3]
```diff
@@ -47,3 +51,65 @@ func (opts *globalOpts) OrgID() string {
 	opts.orgID = config.OrgID()
 	return opts.orgID
 }
+
+// deleteOpts options required when deleting a resource.
+// A command can embed this structure and then safely rely on the methods Confirm, DeleteFromProject or Delete
+// to manage the interactions with the user
+type deleteOpts struct {
```
Comment (gssbzn) [inline internal/cli/cli.go]: These are the main change and the rest is just deleting the duplicates to use this struct
We should be able now to embed this and reuse the Confirm, DeleteFromProject or Delete methods

File: internal/cli/atlas_clusters_delete.go [hunk 4]
```diff
@@ -26,53 +23,32 @@ import (
 
 type atlasClustersDeleteOpts struct {
 	*globalOpts
-	name    string
-	confirm bool
-	store   store.ClusterDeleter
+	*deleteOpts
+	store store.ClusterDeleter
```
Comment (colm-quinn) [inline internal/cli/atlas_clusters_delete.go]: 👍

**Pull request merged**